src/ciphersuite/mod.rs

```diff
@@ -369,17 +349,14 @@ impl AeadKey {
         msg: &[u8],
         aad: &[u8],
         nonce: &AeadNonce,
-    ) -> Result<Vec<u8>, AEADError> {
-        let (ct, tag) = match aead_encrypt(
+    ) -> Result<Vec<u8>, AeadError> {
```

Comment:
Just from a naming standpoint it's not clear why an `AeadError` would not be a `CryptoError`. Maybe convert?

Reply:
I don't want to wrap any crypto errors in a different crypto error (what would be the point?). `CryptoError` is really only for cases where something in the crypto library goes wrong that shouldn't go wrong. Everything else are "valid" cases that can occur.